lib/DataIntegrityProof.js

```diff
@@ -164,19 +166,26 @@ export class DataIntegrityProof extends LinkedDataProof {
 
     // set default `now` date if not given in `proof` or `options`
     let date = this.date;
-    if(proof.created === undefined && date === undefined) {
-      date = new Date();
-    }
+    if(date === null) {
+      if(proof.created !== undefined) {
+        delete proof.created;
+      }
+    } else {
+      if(proof.created === undefined && date === undefined) {
+        date = new Date();
+      }
```

Comment:
So I think we likely don't need any changes here in `createProof`; we don't want to delete `proof.created` if it was specified in `proof`, we just don't want to add something to it if `date` is `null`. It may then follow that we don't need to change anything else here.

Reply:
So what I mean is if I checkout this branch and *just* make the one change in the constructor that sets `this.date = null` (when `date === null`), then the tests pass.